homeassistant/components/binary_sensor/template.py

```diff
@@ -172,6 +185,20 @@ def _async_render(self):
                 return
             _LOGGER.error("Could not render template %s: %s", self._name, ex)
 
+        if self._availability_template:
+            try:
+                self._available = self._availability_template.async_render(
+                ).lower() == 'true'
+            except TemplateError as ex:
+                if ex.args and ex.args[0].startswith(
+                        "UndefinedError: 'None' has no attribute"):
+                    # Common during HA startup - so just a warning
+                    _LOGGER.warning("Could not render template %s, "
+                                    "the state is unknown", self._name)
+                    return
```

Comment:
If the availability template fails to render, the other icon and entity picture templates will not be evaluated because of this line.

Can't you just make this a part of the for loop below?

Reply:
@OttoWinter yeah, I just copied the value_template logic but thinking about it I'm unsure if failing to render the availability_template should always set available to false or instead ignore the error keeping the current available value. Perhaps the former is better (and more consistent with the home assistant logic where the unavailable states has the precedence over any entity state)